### PR TITLE
[SPIR-V] Generate DebugEntryPoint for Vulkan DebugInfo

### PR DESCRIPTION
**Pull request by greg-lunarg**

### CODE REVIEW

**Comment by greg-lunarg** [comment]
@cassiebeckley Could you please review when it clears CI?

**Comment by cassiebeckley** [comment]
Sure, looking at it now.

**Review by cassiebeckley — APPROVED** [review]
Looks good to me I think, just a couple typos

File: tools/clang/lib/SPIRV/SpirvEmitter.cpp [hunk 1]
```diff
@@ -1303,6 +1304,17 @@ void SpirvEmitter::doFunctionDecl(const FunctionDecl *decl) {
       // Create wrapper for the entry function
       if (!emitEntryFunctionWrapper(decl, func, debugFunction))
         return;
+      // Generate DebugEntryPoint if function defintion
```
[the remark this answers is not in the document]
Comment (greg-lunarg) [reply tools/clang/lib/SPIRV/SpirvEmitter.cpp]: Fixed

File: tools/clang/lib/SPIRV/SortDebugInfoVisitor.cpp [hunk 2]
```diff
@@ -55,6 +55,14 @@ void SortDebugInfoVisitor::whileEachOperandOfDebugInstruction(
     if (!visitor(inst->getDebugFunction()))
       break;
   } break;
+  case SpirvInstruction::IK_DebugEntryPoint: {
+    SpirvDebugEntryPoint *inst = +dyn_cast<SpirvDebugEntryPoint>(di);
```
[the remark this answers is not in the document]
Comment (greg-lunarg) [reply tools/clang/lib/SPIRV/SortDebugInfoVisitor.cpp]: Fixed

**Comment by greg-lunarg** [comment]
Thank you!

**Pull request merged**